src/scales/scale.time.js

```diff
@@ -477,33 +507,31 @@ module.exports = function(Chart) {
 			me.max = Math.max(min + 1, max);
 
 			// PRIVATE
-			me._datasets = timestamps;
+			me._datasets = datasets;
 			me._horizontal = me.isHorizontal();
-			me._labels = labels.sort(sorter);    // Sort labels **after** data have been converted
+			me._timestamps = timestamps;
```

Comment:
It's not all the data though (just the timestamps), so I think that might make it harder to read. Maybe `_dataMillis`, `_labelMillis`, and `_datasetMillis` ?